### ISSUE TITLE
Failing isinstance check in tensorflow.python.keras.models._clone_functional_model

### ISSUE THREAD

**Issue by sytham** [tensorflow/tensorflow#23216]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow): No
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
CentOS, specifically
Linux 3.10.0-693.5.2.el7.x86_64 #1 SMP Fri Oct 20 20:32:50 UTC 2017 x86_64 x86_64 x86_64 GNU/Linux

- TensorFlow installed from (source or binary): Through conda
- TensorFlow version (use command below): 1.11.0
- Python version:
Python 2.7.15 |Anaconda, Inc.| (default, Oct 10 2018, 21:32:13)
[GCC 7.3.0] on linux2
-Bazel version: NA
-CUDA/cuDNN version: NA
-GPU model and memory: NA
-Exact command to reproduce: NA
-Mobile device: NA

**Describe the current behavior**
`tensorflow.python.keras.models._clone_functional_model `raises a `ValueError `because the model passed to it is supposedly not an instance of `Model`, even though it is.

**Code to reproduce the issue**
```
from tensorflow import keras
from keras.models import Model
from tensorflow.python.keras.engine import training
Model2 = training.Model
isinstance(Model(), Model2)
```
This returns False. Even though `keras.models.Model` should just be an alias for `tensorflow.python.keras.engine.training.Model`.

**Other info / logs**
I run into this issue when calling` tensorflow.python.keras.estimator.model_to_estimator`.
I can work around it by importing all my keras imports using the full `tensorflow.python.keras...., `but I'm not sure why that should be necessary.

**Comment by tensorflowbutler** [comment]
Thank you for your post. We noticed you have not filled out the following field in the issue template. Could you update them if they are relevant in your case, or leave them as N/A? Thanks.
Bazel version
CUDA/cuDNN version
GPU model and memory
Exact command to reproduce
Mobile device

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 14 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by tensorflowbutler** [comment]
Nagging Assignee @fchollet: It has been 29 days with no activity and this issue has an assignee. Please update the label and/or status accordingly.

**Comment by rmothukuru** [comment]
@sytham,
This issue has been resolved. Please find [the Gist](https://colab.research.google.com/gist/rmothukuru/cf5a882c7db7df102219ac74d8bb7b52/gh_23216.ipynb) of the working code. Thanks!
